pkg/common/controlprotocol/controlprotocol_test.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controlprotocol
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	ctrl "knative.dev/control-protocol/pkg"
+	"knative.dev/control-protocol/pkg/network"
+	ctrlservice "knative.dev/control-protocol/pkg/service"
+
+	"knative.dev/eventing-kafka/pkg/common/controlprotocol/commands"
+	ctrltesting "knative.dev/eventing-kafka/pkg/common/controlprotocol/testing"
+)
+
+func TestNewServerHandler(t *testing.T) {
+
+	saveStartServer := startServerWrapper
+	defer func() { startServerWrapper = saveStartServer }()
+
+	for _, testCase := range []struct {
+		name      string
+		serverErr error
+	}{
+		{
+			name: "No Error",
+		},
+		{
+			name:      "Server Error",
+			serverErr: fmt.Errorf("test error"),
+		},
+	} {
+		t.Run(testCase.name, func(t *testing.T) {
+			mockService := &ctrltesting.MockControlProtocolService{}
+			mockService.On("MessageHandler", mock.Anything).Return()
+			startServerWrapper = func(_ context.Context, _ ...network.ControlServerOption) (*network.ControlServer, error) {
```

Comment:
not asserting expectations on `mockService` ?

Reply:
Ach, I completely forgot about that when I converted the mocks to testify-style.  I will go through the tests and make sure they're doing that for all the ones I added .On()